core-bundle/src/Messenger/WebWorker.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Messenger;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Output\NullOutput;
+use Symfony\Component\EventDispatcher\Attribute\AsEventListener;
+use Symfony\Component\HttpKernel\Event\TerminateEvent;
+use Symfony\Component\Messenger\Command\ConsumeMessagesCommand;
+use Symfony\Component\Messenger\Event\WorkerRunningEvent;
+use Symfony\Component\Messenger\Event\WorkerStartedEvent;
+
+/**
+ * This service accepts an array of Symfony Messenger transports and automatically
+ * detects which of those have real workers running. In case they do not, it falls
+ * back to a web worker (kernel.terminate event) to ensure, messages are always
+ * processed, no matter whether a real worker is running or not.
+ *
+ * Detecting works by using the WorkerStartedEvent and WorkerRunningEvent which
+ * sets/updates a cache items and allowing a grace period. The grace period defines how
+ * long after the last event the web worker should consider a real worker to be running.
+ * Imagine you only have one worker running, and it works on a message for 10 minutes.
+ * In such a case, no WorkerStartedEvent or WorkerRunningEvent will be triggered for 10
+ * minutes so if there were a grace period of only a few seconds, the web worker would
+ * jump in probably too quickly.
+ *
+ * In any case, this provides advantages as it allows us to always use a queue and
+ * postpone processes to at least kernel.terminate in case there are no real
+ * workers available!
```

Comment:
```suggestion
 * This service accepts an array of Symfony Messenger transports and automatically
 * detects which ones have real workers running. In case they have not, it falls
 * back to a web worker at "kernel.terminate" to ensure that messages are always
 * processed, no matter whether a real worker is running or not.
 *
 * Detecting works by using the WorkerStartedEvent and WorkerRunningEvent which
 * sets/updates a cache item and allows a grace period. The grace period defines
 * how long after the last event the web worker should consider a real worker to
 * be running. Imagine you only have one worker running, and it works on a message
 * for 10 minutes. In such a case, no WorkerStartedEvent or WorkerRunningEvent
 * will be triggered for 10 minutes, so if there were a grace period of only a few
 * seconds, the web worker would probably jump in too quickly.
 *
 * In any case, this allows us to always use a queue and postpone processes to at
 * least the "kernel.terminate" event if there are no real workers available!
```